docs/ansible/roles/ansible_plugins/getting-started.rst

```diff
@@ -39,21 +39,98 @@ Jinja templates:
   This filter plugin can be used to filter strings or lists that match shell
   glob patterns.
 
+``split``
+  This filter plugin can be used to split strings, similarly to the
+  ``.split()`` function in Python.
+
+
+.. _ansible_plugins_config_filters:
+
+Configuration filters
+~~~~~~~~~~~~~~~~~~~~~
+
+These filters are used to implement DebOps :ref:`universal_configuration`.
+See the user-facing documentation for the behaviors they are meant to
+facilitate.
+
 ``parse_kv_config``
   Parse a YAML list of dictionaries and output a sorted and expanded list of
   YAML dictionaries that contain a common set of dictionary keys. The filter
   supports dynamic order of the entries using weight model, and can be used to
   generate a configuration file which uses a key/value syntax with unique keys.
 
+  The ``parse_kv_config`` filter accepts this argument:
+
+  ``name``
+    Optional, String. Defaults to ``name``.
+    Sets the name of the field to be used as the unique key.
+
+
 ``parse_kv_items``
   This is a wrapper for the ``parse_kv_config`` filter which can be used in the
   looped Ansible tasks to manage multiple files with key/value syntax, or
   generate a configuration file with multiple key/value configuration
   structures.
 
-``split``
-  This filter plugin can be used to split strings, similarly to the
-  ``.split()`` function in Python.
+  The ``parse_kv_items`` filter accepts the following arguments:
+
+  ``name``
+    Optional, String. Defaults to ``name``.
+    Sets the name of the field to be used as the unique key.
+
+  ``empty``
+    Optional, Mapping. Keys are fields which might be empty, values
+    are other field names or lists of field names.
+
+    The value of the first field with a value other than ``None`` will be used
+    as the value of the specified field, if the specified field is empty.
+
+    This behavior does not extend to fields in second-level lists, such as
+    ``options`` or other defined ``merge_keys``.
+
+    Examples:
+
+    .. code-block:: python
+
+      empty={
+        'some_param':  'other_param',
+        'empty_param': ['param2', 'param1']
+      }
```

Comment:
I would stick to the YAML syntax here as this is how users will interact with this Python filter.

Reply:
Yes, you're absolutely right, I was on autopilot :)

Reply:
The dictionary needs to be specified like this, but it can be shown better. This should probably be something like...
```yaml
{{ variable | parse_kv_items(empty={
                             'some_param': 'other_param',
                             'empty_param': [ 'param2', 'param1']}) }}
```

Reply:
I'm expanding the example with results as well, I had trouble understanding it now even though I did when I wrote it!  :D

Reply:
Um, actually... the usage I see in the codebase are inline python supplied in `parse_kv_items(args)`... I guess the items could be built as yaml but I think python is the most direct usage. What do you think?

Reply:
I think something like:
```python
parse_kv_items(
  empty={
    'some_param':  'other_param',
    'empty_param': ['param2', 'param1']
  }
)
```

```yaml
- name: foo
  other_param: bar

# Will result in

- name: foo
  some_param: bar
  other_param: bar
```

Reply:
@drybjed do you want me to wrap the ``parse_kv_items`` example in jinja? I only now saw your comment.

Reply:
@tasosalvas This is how the example will most likely be used within Ansible roles, so I would say wrapping it in Jinja tags should help explain the intended usage.

Reply:
Colorization sure looks prettier this way

Reply:
I was wrong. I attribute that to being tried when I reviewed this. Thanks for taking the opportunity to improve the docs even further!